code-generator/core/src/main/java/com/tigrisdata/tools/schema/TigrisDBObjectRule.java

```diff
@@ -275,10 +277,24 @@ private JDefinedClass createClass(String nodeName, JsonNode node, JPackage _pack
                   ruleFactory.getNameHelper().getUniqueClassName(nodeName, node, _package),
                   ClassType.CLASS);
         } else {
-          String collectionName = node.get("name").asText();
-          String javaTypeName =
-              Inflector.getInstance().singularize(StringUtils.capitalize(collectionName));
+          String className = StringUtils.capitalize(nodeName);
+          String collectionName = null;
+          // if it is the sub object
+
+          if (node.has("name")) {
```

Comment:
So if the node doesn't have a name then we should error out?

Reply:
added shallow validation. I think we will need a layer in front here to add full validation and keep in sync with server.